client/landing/stepper/declarative-flow/internals/hooks/use-step-navigation-with-tracking/index.ts

```diff
@@ -54,11 +54,13 @@ export const useStepNavigationWithTracking = ( {
 		() => ( {
 			...( stepNavigation.submit && {
 				submit: ( providedDependencies: ProvidedDependencies = {}, ...params: string[] ) => {
-					handleRecordStepNavigation( {
-						event: STEPPER_TRACKS_EVENT_STEP_NAV_SUBMIT,
-						providedDependencies,
-						additionalProps: tracksEventPropsFromFlow?.[ STEPPER_TRACKS_EVENT_STEP_NAV_SUBMIT ],
-					} );
+					if ( ! providedDependencies?.skipSubmitStepTracks ) {
```

Comment:
What do you think of calling it `shouldSkipSubmitTracking`?

Maybe you can simplify this if with: 

```js
! providedDependencies?.skipSubmitStepTracks &&
	handleRecordStepNavigation( {
		event: STEPPER_TRACKS_EVENT_STEP_NAV_SUBMIT,
		providedDependencies,
		additionalProps: tracksEventPropsFromFlow?.[ STEPPER_TRACKS_EVENT_STEP_NAV_SUBMIT ],
	} );

```